streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -1283,6 +1283,18 @@ object ZSink extends ZSinkPlatformSpecific with Serializable {
   final def drain: ZSink[Any, Nothing, Nothing, Any, Unit] =
     foldLeft(())((s, _) => s)
 
+  /**
+   * Creates a sink containing the first value.
+   */
+  final def head[A]: ZSink[Any, Nothing, A, A, Option[A]] =
+    foldUntil[Option[A], A](None, 1) { case (_, a) => Some(a) }
+
+  /**
+   * Creates a sink containing the last value.
+   */
+  final def last[A]: ZSink[Any, Nothing, A, A, Option[A]] =
```

Comment:
There is no remainder so you should use `Nothing` for the `A0` slot. This will improve type inference, too.

Reply:
I tried that, but using `ZSink.last` instead of  e.g. `ZSink.last[Int]` still results in the same compilation error as before.